doc/conf.py

```diff
@@ -78,131 +79,135 @@
-# unit titles (such as .. function::).
-#add_module_names = True
-
-# If true, sectionauthor and moduleauthor directives will be shown in the
-# output. They are ignored by default.
-#show_authors = False
+# pygments_style = 'sphinx'
 
-# The name of the Pygments (syntax highlighting) style to use.
-pygments_style = 'sphinx'
+todo_include_todos = False
 
-# A list of ignored prefixes for module index sorting.
-#modindex_common_prefix = []
+html_baseurl = 'https://contrib.scikit-learn.org/skglm/'
 
-# If true, keep warnings as "system message" paragraphs in the built documents.
-#keep_warnings = False
+extensions.append("sphinxext.opengraph")
 
-# If true, `todo` and `todoList` produce output, else they produce nothing.
-todo_include_todos = False
+# OpenGraph config
+ogp_site_url = html_baseurl
+ogp_image = "https://contrib.scikit-learn.org/skglm/_static/images/logo.svg"
+ogp_description_length = 250
+ogp_type = "website"
 
 
 # -- Options for HTML output ----------------------------------------------
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'furo'
+html_theme = 'pydata_sphinx_theme'
+
+dev_color = "#ff7f0e"
+stable_color = "#1f77b4"
+
+primary = stable_color if is_stable_doc else dev_color
+primary_dark = "#135b91" if is_stable_doc else "#e76f00"
+
+version_match = "stable" if is_stable_doc else "dev"
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
 html_theme_options = {
+    # color configurations
     "light_css_variables": {
-        "color-brand-primary": "#991b1b",
-        "color-brand-content": "#991b1b",
-        "font-stack": 'ui-sans-serif, system-ui',
-        "color-background-secondary": "#fafafa"
+        "color-brand-primary": primary,
+        "color-brand-content": primary,
+        "color-brand-secondary": primary_dark,
     },
     "dark_css_variables": {
-        "color-brand-primary": "#fca5a5",
-        "color-brand-content": "#fca5a5",
+        "color-brand-primary": primary_dark,
+        "color-brand-content": primary_dark,
+        "color-brand-secondary": primary,
     },
+
+    # Navbar configuration
+    "navbar_align": "left",
+    "navbar_start": ["navbar-logo"],
+    "navbar_center": ["navbar-nav"],
+    "navbar_end": [
+        "navbar-icon-links",
+        "theme-switcher",
+        "version-switcher",
+    ],
+
+    # other configurations
+    "icon_links": [
+        {
+            "name": "GitHub",
+            "url": "https://github.com/scikit-learn-contrib/skglm",
+            "icon": "fab fa-github",
+            "type": "fontawesome",
+        },
+    ],
+    "version_dropdown": True,
+    "switcher": {
+        # NOTE: must be a URL and not a path relative to doc
+        "json_url": "https://raw.githubusercontent.com/floriankozikowski/skglm/refs/heads/documentation-update/doc/_static/switcher.json",
```

Comment:
the `json_url` must be a URL not a path in doc folder 
c.f. https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/version-dropdown.html#configure-switcher-json-url

once we merge this PR, we will change it,
this one is a work-around to check that the switcher renders/works as expected